src/main/webapp/app/exercises/programming/manage/update/update-components/language/programming-exercise-language.component.html

```diff
@@ -0,0 +1,240 @@
+        jhiTranslate="artemisApp.programmingExercise.wizardMode.detailedSteps.languageStepTitle"
+        id="artemisApp.programmingExercise.wizardMode.detailedSteps.languageStepTitle"
+    ></h3>
+    <p jhiTranslate="artemisApp.programmingExercise.wizardMode.detailedSteps.languageStepMessage"></p>
+    @if (isEditFieldDisplayedRecord().programmingLanguage) {
+        <div class="form-group">
+            <label class="label-narrow" jhiTranslate="artemisApp.programmingExercise.programmingLanguage" for="field_programmingLanguage"></label>
+            <select
+                #select="ngModel"
+                required
+                class="form-select"
+                [ngModel]="programmingExerciseCreationConfig.selectedProgrammingLanguage"
+                (ngModelChange)="select.model = programmingExerciseCreationConfig.onProgrammingLanguageChange($event)"
+                name="programmingLanguage"
+                id="field_programmingLanguage"
+                [disabled]="programmingExerciseCreationConfig.isImportFromExistingExercise || !!programmingExercise.id || programmingExerciseCreationConfig.isImportFromFile"
+            >
+                @for (supportedLanguage of programmingExerciseCreationConfig.supportedLanguages; track supportedLanguage) {
+                    <option [value]="supportedLanguage" [jhiTranslate]="'artemisApp.ProgrammingLanguage.' + supportedLanguage"></option>
+                }
+            </select>
+        </div>
+    }
+    @if (programmingExercise.programmingLanguage && programmingExerciseCreationConfig.projectTypes?.length && programmingExerciseCreationConfig.modePickerOptions) {
+        <div class="form-group mt-2">
+            @if (isEditFieldDisplayedRecord().projectType) {
+                <label class="label-narrow" jhiTranslate="artemisApp.programmingExercise.projectType" for="field_projectType"> </label>
+                <jhi-mode-picker
+                    required
+                    [options]="programmingExerciseCreationConfig.modePickerOptions"
+                    [value]="programmingExerciseCreationConfig.selectedProjectType"
+                    (valueChange)="programmingExerciseCreationConfig.onProjectTypeChange($event!)"
+                    [disabled]="programmingExerciseCreationConfig.isImportFromExistingExercise || programmingExerciseCreationConfig.isImportFromFile || !!programmingExercise.id"
+                    name="projectType"
+                    id="field_projectType"
+                />
+            }
+            @if (
+                isEditFieldDisplayedRecord().withExemplaryDependency &&
+                !programmingExerciseCreationConfig.isImportFromExistingExercise &&
+                !programmingExerciseCreationConfig.isImportFromFile &&
+                !programmingExercise.id &&
+                programmingExercise.programmingLanguage === ProgrammingLanguage.JAVA &&
+                programmingExercise.projectType !== ProjectType.MAVEN_BLACKBOX
+            ) {
```

Comment:
That is a rather hard-to-understand if condition.

Reply:
I think that this is out of scope for this PR, I do not think that this can easily be changed as `programmingExerciseCreationConfig` is not a signal in the parent component.
This would be a rather big refactoring, that once done, allows to use computed signals instead.